collectors/ApicLeafCapacityCollector.py

```diff
@@ -37,45 +39,84 @@ def get_metrics(self, host: str, data: Dict) -> List[GaugeMetricFamily]:
+                            g_leaf_cap_tcam.add_metric(labels=[leaf_id, 'local', 'l2'],
+                                                       value=l2_local)
+                        elif 'eqptcapacityL2RemoteUsage5min' in data_object:
+                            l2_remote = data_object['eqptcapacityL2RemoteUsage5min']['attributes']['remoteEpLast']
+                            g_leaf_cap_tcam.add_metric(labels=[leaf_id, 'remote', 'l2'],
+                                                       value=l2_remote)
+                else:
+                    for data_object in leaf_data:
+                        if 'eqptcapacityL3RemoteUsageCap5min' in data_object:
+                            l3_max = data_object['eqptcapacityL3RemoteUsageCap5min']['attributes']['v4RemoteEpCapMax']
+                            g_leaf_cap_tcam.add_metric(labels=[leaf_id, 'max', 'l3'],
+                                                       value=l3_max)
+                        elif 'eqptcapacityL3Usage5min' in data_object:
+                            l3_local = data_object['eqptcapacityL3Usage5min']['attributes']['v4LocalEpLast']
+                            g_leaf_cap_tcam.add_metric(labels=[leaf_id, 'local', 'l3'],
+                                                       value=l3_local)
+                        elif 'eqptcapacityL3RemoteUsage5min' in data_object:
+                            l3_remote = data_object['eqptcapacityL3RemoteUsage5min']['attributes']['v4RemoteEpLast']
+                             g_leaf_cap_tcam.add_metric(labels=[leaf_id, 'remote', 'l3'],
+                                                       value=l3_remote)
+                        elif 'eqptcapacityL2Usage5min' in data_object:
+                            l2_max = data_object['eqptcapacityL2Usage5min']['attributes']['localEpCapMax']
+                            g_leaf_cap_tcam.add_metric(labels=[leaf_id, 'max', 'l2'],
+                                                       value=l2_max)
+                            l2_local = data_object['eqptcapacityL2Usage5min']['attributes']['localEpLast']
+                            g_leaf_cap_tcam.add_metric(labels=[leaf_id, 'local', 'l2'],
+                                                       value=l2_local)
+                        elif 'eqptcapacityL2RemoteUsage5min' in data_object:
+                            l2_remote = data_object['eqptcapacityL2RemoteUsage5min']['attributes']['remoteEpLast']
+                            g_leaf_cap_tcam.add_metric(labels=[leaf_id, 'remote', 'l2'],
+                                                       value=l2_remote)
+                        l3_total = l3_local + l3_remote
+                        l2_total = l2_local + l2_remote
+                        g_leaf_cap_tcam.add_metric(labels=[leaf_id, 'total', 'l3'], value=l3_total)
+                        g_leaf_cap_tcam.add_metric(labels=[leaf_id, 'total', 'l2'], value=l2_total)
```

Comment:
Is the indentation right at these lines? Right now this will be executed at every data_object. I suppose the `l3_total` and `l2_total` should be initialized before `leaf_data` is iterated over.

Reply:
every node need these two values so I think it's correct

Reply:
That would be the case if you move them out of this loop. Currently it would be calculated for every entry in `leaf_data` instead once per `leaf`.

Reply:
Just wondering should this only be calculated for the non-gen1 leaf switches or for both?